src/modules/user/services/user.service.ts

```diff
@@ -72,6 +78,78 @@ export class UserService {
     });
   }
 
+  public async saveCardInfo({
+    id,
+    step,
+    cardDto,
+  }: {
+    id: string;
+    step: number;
+    cardDto: PaymentMethod;
+  }): Promise<void> {
+    const stripe = new Stripe(this.configService.get<string>('STRIPE_SECRET'));
+
+    const customer = await stripe.customers.create({
+      payment_method: cardDto.id,
+      email: cardDto.billing_details.email,
+      invoice_settings: {
+        default_payment_method: cardDto.id,
+      },
+    });
+
+    const user = await this.userRepository.findOneBy({ id });
```

Comment:
mb move this check at start? so we dont need even to call stripe if user doesnt exist?